### ISSUE TITLE
[Feature Request] Backup game folder/wine prefix to archive

### ISSUE THREAD

**Issue by Julian2000nl** [lutris/lutris#1451]
When right-clicking a game, make an option "Back up to archive" which copies the game folder(with the drive_c and Wine prefix) to a date labelled .tar.gz archive.
Let's say you run a game using Wine/DXVK and you want to update it or install mods and you mess everything up(ex. Install component via Winetricks that breaks stuff) then you have a backup of a working version. Would really be helpful given the fact a lot of games are ~100GB nowadays and re-installing them from scratch can be very time consuming.

**Comment by openglfreak** [comment]
Squashfs is better. Fight me.

**Comment by founderio** [comment]
Hint: some wine games store essential data in a user folder inside the prefix. It may be sensible to **optionally** adjust the name of that folder upon unpacking. This is useful for transferring game installations between computers.

**Comment by telyn** [comment]
It'd be very useful if the backup also contained all the Lutris configuration for the application too. Arguments to the executable, which version of wine to use, etc. could all be quite important to making the game archive usable in the longer-term.

**Comment by strycore** [comment]
This is mostly completed. Still needs a bit more QA

**Comment by teotikalki** [comment]
This would be great. I've found that, aside from League of Legends, pretty much every game I play is moddable. Anyone who is into modding will know that you can spin a delicate web of version dependencies. Upgrading to a new game version breaks your game because mods haven't updated, and sometimes the mods DON'T update because the original author doesn't actually play the game anymore or whatnot. The ability to backup something like Terraria so as to see how a new version plays with my old mods would be FANTASTIC; I'd actually be a LOT more likely to actually update games if I could easily roll back the update should it turn out to be incompatible.

It would also be nice for when someone pulls an Owlcat and inserts spyware into a game two years after release and it updates, you launch, and then are told that you have to comply with new terms that would have made you not play the game in the first place. Having a backup that you can roll back to and prevent updating would let you keep the version you're happy with.

In short, it seems like a fantastic features with multiple use cases.

BTW, this is a little off topic, but... Thank you guys so much for Lutris. As far as I'm concerned it's the best thing to happen to gaming on linux, which I think is good for both linux and gaming.

**Comment by kenthinson** [comment]
> This is mostly completed. Still needs a bit more QA

I am willing to help test if you need. Is there a specific branch I need to compile? Also I’m on fedora if that matters at all.

**Comment by strycore** [comment]
We don't have an UI for it yet but you can run 

```
./bin/lutris  --export game-slug --dest=/folder/to/use/for/export
```

**Comment by pglempereur** [comment]
Just had a go with the export ! A couple of things I noticed : 
- if the destination path points to a folder that doesn't exist, the output just says "[game] exported to [path]/[game].tar.xz (but it isn't)
- the export was slow. CPU and disk usage were low. A ~20GB archive took a bit more than 2 hours.

I'm using the Flatpak from Flathub on Fedora 41, if it matters. I used `flatpak run --command=sh net.lutris.Lutris` and then the command you gave.

Thanks for the feature, and for Lutris !

**Comment by kenthinson** [comment]
Okay reporting back. 
So export path only works with a full path. If I use for example ~/Desktop/ to try to send to my desktop it fails. However like mentioned before it says it was exported when it was not.

On the subject of the export time. It would be nice to have flags to tune the desired xz compression amount. Or even have none at all as a option and just export as a raw stream of data. For example I'm exporting to a ZFS file system so it does compression on the file system level. A raw data dump would work better for me. 

I'm going to try and see if I can get lutris compiling on my system maybe I'll do a little hacking on the options. 

@strycore Is there a working import flag also ready for testing?
